src/layouts/Media/Media.tsx

```diff
@@ -63,15 +64,50 @@ const getMediaLabels = (mediaType: "files" | "images"): MediaLabels => {
   }
 }
 
+const MEDIA_PAGINATION_SIZE = 15
+
 export const Media = (): JSX.Element => {
   const history = useHistory()
+  const [curPage, setCurPage] = usePaginate()
   const { params, path, url } = useRouteMatch<{
     siteName: string
     mediaRoom: "files" | "images"
     mediaDirectoryName: string
   }>()
   const { siteName, mediaRoom: mediaType } = params
-  const { data: mediasData, isLoading } = useGetMediaFolders(params)
+
+  useEffect(() => {
+    // NOTE: Because this component is shared between different media types + subfolders,
+    // we need to reset the page number to 1 when the url changes.
+    // If this is not done, we might end up with erroneous pagination
+    // such as being on page 4 of a 1 page list.
+    // This results in the back button being clickable,
+    // but the number always being 1.
+    setCurPage(1)
+    // NOTE: **NOT** adding `setCurPage` as a dependency here
```

Comment:
Could we add an eslint ignore if we are going against the recommended usage?

Reply:
sure